eslint/core/deprecated-alias.ts

```diff
@@ -0,0 +1,159 @@
+import type {Rule} from 'eslint';
+import type ESTree from 'estree';
+
+const traverse = require('eslint-traverse');
+
+const meta = {
+  type: 'suggestion',
+
+  docs: {
+    description: 'replace deprecated aliases',
+    recommended: true,
+  },
+  settings: {
+    'import/parsers': {
+      '@typescript-eslint/parser': ['.ts', '.js', '.tsx', '.jsx']
+    }
+  },
+  fixable: 'code'
+} as Rule.RuleModule['meta'];
+
+export interface ESLintDeprecationCfg {
+  /** Current alias name */
+  alias: string;
+  /** Deprecated name */
+  deprecation: string;
+}
+
+type ImportNode = ESTree.ImportSpecifier & {
```

Comment:
Shortened types a little, but can't get rid of them completely:
for ```type ImportNode = ESTree.ImportSpecifier & Rule.NodeParentExtension;```, the similar type can be found in ESTree namespace, but it doesn't have a shorthand and it's not exported:
```
interface NodeListener {
  //...
  ImportSpecifier?: ((node: ESTree.ImportSpecifier & NodeParentExtension) => void) | undefined;
  //...
}
```
In case of BaseNode, we can't use `NodeParentExtension` type, since it doesn't have `kind` property. So we need to define out own `parent` property